project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/IJDDContext.kt

```diff
@@ -1,9 +1,22 @@
 package org.plan.research.minimization.plugin.model
 
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import com.intellij.openapi.vfs.VirtualFile
 import org.plan.research.minimization.core.model.DDContext
 
-// FIXME: In the current implementation the projects are not closed and disposed which leads to memory leaks.
-//  Need to adjust the architecture to comprehend this.
-//  E.g. fix the moment when the project is closed or replace project with `Path` and open the project each time it needed
-data class IJDDContext(val project: Project) : DDContext
+
+/**
+ * Represents a context for the minimization process containing the current project and derived properties.
+ *
+ * @property project The current project to be minimized.
+ * @property originalProject The original project before any minimization stages, defaults to [project].
+ * @property currentLevel An optional list of [ProjectFileDDItem] representing the current level of the minimizing project files.
+ */
+data class IJDDContext(
+    val project: Project,
+    val originalProject: Project = project,
+    val currentLevel: List<ProjectFileDDItem>? = null,
```

Comment:
It's not important at all, but some getter to get `List<VirtualFile>` for `project` & `currentLevel` (since they're immutable) could be useful